app/src/main/java/com/jerboa/ui/components/post/PostPane.kt

```diff
@@ -133,13 +133,19 @@ fun PostScreen(
     postActionBarMode: PostActionBarMode,
     swipeToActionPreset: SwipeToActionPreset,
 ) {
-    Log.d("jerboa", "got to post screen")
+    Log.d("jerboa", "got to post pane")
 
     val ctx = LocalContext.current
 
     val account = getCurrentAccount(accountViewModel = accountViewModel)
 
-    val postViewModel: PostViewModel = viewModel(factory = PostViewModel.Companion.Factory(id))
+    val postViewModel: PostViewModel = viewModel(factory = PostViewModel.Companion.Factory(postOrCommentId))
+
+    // Some hacky code required to update the screen for tablet view
+    // Necessary because the viewmodel initializes only once.
+    if (postViewModel.id !== postOrCommentId) {
+        postViewModel.reInitializeWithNewId(postOrCommentId)
+    }
```

Comment:
This had me scratching my head for a while. Clicking on a new post in the list didn't re-init the viewmodel, so it kept showing the first post selected.

Reply:
ViewModels are bound to the lifetime of the screen. It reuses it as long as the screen exists.

see https://developer.android.com/topic/libraries/architecture/viewmodel#jetpack-compose